crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -96,14 +98,27 @@ impl Solution {
                             buy: solution.prices
                                 [&fulfillment.order().buy.token.wrap(solution.weth)],
                         };
-                        let fulfillment = fulfillment.with_protocol_fees(prices)?;
-                        trades.push(Trade::Fulfillment(fulfillment))
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade)
                     }
                     order::Kind::Liquidity => {
                         trades.push(trade);
                     }
                 },
-                Trade::Jit(_) => trades.push(trade),
+                Trade::Jit(jit) => {
+                    if cow_amm_addresses.contains(&jit.order().receiver.into()) {
+                        let prices = ClearingPrices {
+                            sell: solution.prices[&jit.order().sell.token.wrap(solution.weth)],
+                            buy: solution.prices[&jit.order().buy.token.wrap(solution.weth)],
+                        };
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade);
+                    } else {
+                        let mut jit = jit.clone();
+                        jit.remove_protocol_fees();
+                        trades.push(Trade::Jit(jit));
+                    }
```

Comment:
Let's simplify this by assuming no jit order has protocol fees (I don't see a use case for that right now and it adds a bunch of complexity)